src/sync/deployment.test.ts

```diff
@@ -30,6 +37,225 @@ describe("sync/deployments", () => {
+				});
+			});
+
+			describe("for existing deployment cursor", () => {
+				// eslint-disable-next-line jest/expect-expect
+				it("should fetch deployments from existing cursor", async () => {
+
+					const deploymentCount = 4;
+					const deployments = createDeploymentEntities(deploymentCount);
+
+					nockFetchingDeploymentgPagesGraphQL(getDeploymentsQuery, deployments[2].cursor, [deployments[1], deployments[0]]);
+
+					nockDeploymentCommitGetApi([deployments[1], deployments[0]], REPEAT_ONCE);
+
+					//because ff is off, it won't find cache in history, so need to call the rest listing api lots of times
+					nockDeploymentListingApi(deployments, REPEAT_LOTS_OF_TIME);
+					nockDeploymentStatusApi(deployments, REPEAT_LOTS_OF_TIME);
+
+					const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, deployments[2].cursor, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+					expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[1], deployments[0]]);
+				});
+			});
+		});
+
+		//--helper funcs--
+		const msgPayload = () => ({
+			jiraHost,
+			installationId: DatabaseStateCreator.GITHUB_INSTALLATION_ID,
+			commitsFromDate: "2023-01-01T00:00:00Z"
+		});
+
+		//size up to 9 entities
+		const createDeploymentEntities = (size: number) => {
+			return "*".repeat(size).split("").map((_, idx) => {
+				const clone = JSON.parse(JSON.stringify(deploymentNodesFixture.data.repository.deployments.edges[0]));
+				clone._seq = idx + 1;
+				clone.cursor = `cursor:${idx + 1}`;
+				clone.node.createdAt = `2023-01-0${idx + 1}T10:00:00Z`;
+				clone.node.updatedAt = `2023-01-0${idx + 1}T10:00:00Z`;
+				clone.node.databaseId = `dbid-${idx + 1}`;
+				clone.node.commitOid = `SHA${idx + 1}`;
+				clone.node.statuses.nodes.forEach(n => {
+					n.createdAt = clone.node.createdAt;
+					n.updatedAt = clone.node.updatedAt;
+				});
+				return clone;
+			});
+		};
+
+		const nockFetchingDeploymentgPagesGraphQL = (query, cursor, deployments) => {
+			githubNock.post("/graphql", { query, variables: { owner: repoSyncState.repoOwner, repo: repoSyncState.repoName, per_page: PAGE_SIZE__TWO_ITEMS, cursor } })
+				.query(true).reply(200, { "data": { "repository": { "deployments": { "edges": deployments } } } });
+		};
+
+		const nockDeploymentListingApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments?environment=prod&per_page=10`)
+					.reply(200, deployments.map((item, idx) => ({
+						id: item.node.databaseId,
+						sha: item.node.commitOid,
+						ref: "random",
+						task: `task for deployment ${idx + 1}`,
+						payload: {},
+						original_environment: item.node.environment,
+						environment: item.node.environment,
+						description: `description for deployment ${idx + 1}`,
+						creato: { login: "test-repo-owner", id: 1, type: "User" },
```

Comment:
a couple of typos after removing the quotes:

- creato -> creator
- production_environmen -> production_environment